swift/swift.py

```diff
@@ -16,12 +16,95 @@
 import importlib
 import importlib.util
 from contextlib import contextmanager
+from dataclasses import dataclass, asdict
+from typing import Any, Iterable, Mapping, Self
 
 from PyQt5.QtCore import QObject, pyqtSlot, Qt
 from PyQt5.QtWidgets import QApplication, QMainWindow, QDockWidget
 
 from swift.bus import Bus
 
+@dataclass
+class AppInfo:
+    """Information required to create an app.
+    
+    Fields:
+        module: Module name in which the app class resides.
+        class_: App class name.

```

Comment:
I think `cls` looks better according to PEP.

In the [documentation](https://peps.python.org/pep-0008/#designing-for-inheritance),
> If your public attribute name collides with a reserved keyword, append a single trailing underscore to your attribute name. This is preferable to an abbreviation or corrupted spelling. (However, notwithstanding this rule, ‘cls’ is the preferred spelling for any variable or argument which is known to be a class, especially the first argument to a class method.)

How about your opinion?

Reply:
What a great review! I couldn't find that, thank you 👍

Reply:
How do you think of accepting "class" as "cls" in `AppInfo.parse()`?
It is convenient but could be confused.

Reply:
Oh, I agree. There is already `cls` variable..

Reply:
Oh, I changed "class_" -> "cls". I think you misunderstood my question.

I just want your opinion about the current decision of `AppInfo.parse()`.
It takes "class" as well as "cls" (previously "class_"), since the previous setup.json file uses "class".
My question is: should it allow "class" or not (i.e., strictly allow "cls" only)?

Reply:
Umm.. Do you want to set a variable name as "class"?
Is is possible even if "class" is a reseved word?

Reply:
No, the field name is "cls".

However, we give a dictionary as a setup information.
Its keys may contain "class".
Should this be accepted as "cls"?

Reply:
To understand easily, I think we should only allow `cls`.

Thus, please change it and update `setup.json` too.